src/main/java/games/strategy/thread/ThreadPool.java

```diff
@@ -39,10 +37,12 @@ public void runTask(final Runnable task) {
    */
   public void waitForAll() {
     while (!futuresStack.isEmpty()) {
-      if (futuresStack.peek().isDone()) {
-        futuresStack.pop();
-      } else {
-        ThreadUtil.sleep(5);
+      try {
+        futuresStack.pop().get();
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
+      } catch (ExecutionException e) {
+        ClientLogger.logError(e);
```

Comment:
**Minor:** `ExecutionException` offers no additional information; please just log the cause.

**Possibly Not Minor:** We are changing behavior here.  Whereas before, if a task threw an exception, this method would do nothing with it.  Now we'll trigger the Error Console to open.  That is probably the right thing to do, but I just want to ensure we're all in agreement here.